go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -93,6 +98,34 @@ func (a *PodAllocator) Init() error {
 	return nil
 }
 
+// getActiveNetworkForNamespace returns the active network for the given namespace
+// and is a wrapper around util.GetActiveNetworkForNamespace
+func (a *PodAllocator) getActiveNetworkForNamespace(namespace string) (string, error) {
+	return util.GetActiveNetworkForNamespace(namespace, a.nadLister)
+}
+
+// isPrimaryNetwork returns if pod's primary network is same
+// as this controller's network
+func (a *PodAllocator) isPrimaryNetwork(pod *corev1.Pod) (bool, error) {
+	if !util.IsNetworkSegmentationSupportEnabled() {
+		// if user defined network segmentation is not enabled
+		// then we know pod's primary network is "default" and
+		// pod's secondary network is NOT its primary network
+		return !a.netInfo.IsSecondary(), nil
```

Comment:
I would check if it's default so we don't have to negate stuff 
```golang
return a.netInfo.GetNetworkName() == types.DefaultNetworkName
```
If we don't want to use GetNetworkName, we can always add a `IsDefault()` to NetInfo that returns true at DefaultNetInfo.
So we will have

```golang
return a.netInfo.IsDefault()
```

Reply:
done